### ISSUE TITLE
How can we check if we have a valid cookie or not without a popup?

### ISSUE THREAD

**Issue by ceremcem** [nolanlawson/pouchdb-authentication#107]
I think we can use `db.getSession()` or `db.info()` to check whether we have a valid cookie or not, but it creates a "basic authentication popup". `{skipSetup: true}` does not prevent this behaviour. 

How can we check if we have a valid cookie (and/or login with a token) without triggering that popup?

**Comment by nolanlawson** [comment]
That's a good question &ndash; I didn't even realize it was an issue.

Note that pouchdb-authentication is a very thin layer over CouchDB authentication, which uses basic cookie authentication. There is probably some way to do this, but I'm not aware of it.

**Comment by ceremcem** [comment]
I have been searching a way out but it seems that `WWW-Authentication: Basic ...` causes this popup in the server response. As far as I understand, there is nothing we can do without modifying the response and editing/deleting this header.

**Comment by nolanlawson** [comment]
Yes, that header is _not_ cookie authentication. You shouldn't use it if you want to avoid the popup, i.e. you shoudn't pass it in to the `PouchDB` constructor.

**Comment by ceremcem** [comment]
I use Cloudant.com, it's out of my hands... Thank you for your attention...

**Comment by ceremcem** [comment]
Here is a workaround, you may consider merging with pouchdb-authentication: 

```ls
$.ajax do
    type: \GET
    url: 'https://demeter.cloudant.com/_session'
    xhrFields: {+withCredentials}
    headers:
        'Content-Type':'application/x-www-form-urlencoded'
    success: ->
        console.log "We are already logged in!"
        after-logged-in!
    error: -> 
        console.log "We are not logged in yet..."
```

```js
$.ajax({
  type: 'GET',
  url: 'https://demeter.cloudant.com/_session',
  xhrFields: {
    withCredentials: true
  },
  headers: {
    'Content-Type': 'application/x-www-form-urlencoded'
  },
  success: function(){
    console.log("We are already logged in!");
    return afterLoggedIn();
  },
  error: function(){
    return console.log("We are not logged in yet...");
  }
});
```

**Comment by nolanlawson** [comment]
@ceremcem Can you open a pull request please?

**Comment by ceremcem** [comment]
Well, I would love to, but I'm not sure how to integrate this with `utils.ajax()` function. Let me try in this week.

**Comment by ptitjes** [comment]
@ceremcem What is different in the code you showed in comment https://github.com/pouchdb-community/pouchdb-authentication/issues/107#issuecomment-231596708 from what pouchdb-authentication does ? Is it the `withCredentials: true`, the `'Content-Type'` declaration, or both ?

**Comment by ceremcem** [comment]
As far as I can recall, it was only `withCredentials: true` option, but the source code at that time will say more...

**Comment by ptitjes** [comment]
I see. So, IIRC you can pass `{ajax: {withCredentials: true}}` everywhere in the API where you can give options since https://github.com/pouchdb/pouchdb/commit/efe5dd8aa645e349a46a8e293703310c8da06ea2. That means you can do:
```js
db.getSession({ajax: {withCredentials: true}});
```
However, it seems looking at the code that the default value is `true` :/

I think I have to find a way to setup a test case with a CouchDB instance with basic authentication then...

**Comment by Samhamsam** [comment]
Is there a solution for this problem already? If i set WWW-Authenticate to Basic realm="Welcome to the Couch!" i get this problem.
